.github/workflows/release.yml

```diff
@@ -620,28 +649,68 @@ jobs:
             -H "Content-Type: application/json; charset=utf-8" \
             -d "$(jq -n --arg channel "#xrpl-js" --arg text "$text" '{channel:$channel, text:$text}')"
 
-      - name: Notify Slack if tests fail
-        if: failure()
+  generate-documentation:
+    name: Generate and Publish documentation for ${{ needs.get_version.outputs.package_version }}
+    if: ${{ needs.get_version.outputs.is_beta != 'true' }}
+    uses: ./.github/workflows/generate-documentation.yml
+    needs: [get_version, release]
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    with:
+      git_ref: ${{ github.ref_name }}
+
+  notify_failures:
+    runs-on: ubuntu-latest
+    needs:
+      [
+        get_version,
+        run_faucet_test,
+        run_tests,
+        pre_release,
+        ask_for_dev_team_review,
+        first_review,
+        ask_for_sec_team_review,
+        release,
+        generate-documentation
+      ]
+    if: >-
+      ${{ always() && (
+        needs.get_version.result == 'failure' ||
+        (needs.run_faucet_test.result == 'failure' && needs.get_version.outputs.is_beta != 'true') ||
+        (needs.run_tests.result == 'failure' && needs.get_version.outputs.is_beta != 'true') ||
+        needs.pre_release.result == 'failure' ||
+        needs.ask_for_dev_team_review.result == 'failure' ||
+        needs.first_review.result == 'failure' ||
+        needs.ask_for_sec_team_review.result == 'failure' ||
+        needs.release.result == 'failure' ||
+        needs.generate-documentation.result == 'failure'
+      ) }}
+    steps:
+      - name: Notify Slack about workflow failure
         env:
           SLACK_TOKEN: ${{ secrets.SLACK_TOKEN }}
+          PACKAGE_NAME: ${{ github.event.inputs.package_name }}
+          PACKAGE_VERSION: ${{ needs.get_version.outputs.package_version }}
+          NEEDS_JSON: ${{ toJson(needs) }}
         run: |
-          MESSAGE="❌ Release failed for ${{ env.PACKAGE_NAME }}@${{ env.PACKAGE_VERSION }}. Check the logs: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
-          curl -X POST https://slack.com/api/chat.postMessage \
+          set -euo pipefail
```

Comment:
The `notify_failures` job is not pinned to run on any specific shell. However, the pipefail option is configured to work only on `bash` and `ksh` shells. It will not work on `dash` (or) `sh` shells.

Discussion: https://askubuntu.com/questions/886537/set-e-o-pipefail-not-working-on-bash-script-on-ubuntu-16

Some of the jobs in this workflow use `bash` as the pinned shell, is that sufficient for all the jobs?